op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,138 @@
+package upgrades
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/lmittmann/w3"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInteropUpgradeTransactions(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+
+	// Run Isthmus until the interop upgrade
+	is := dsl.SetupInterop(t, dsl.SetInteropOffset(uint64(15)))
+	actors := is.CreateActors()
+	actors.PrepareChainState(t)
+
+	////////////////////////////
+	// Pre-upgrade Block Production
+	////////////////////////////
+
+	fmt.Println("Pre-upgrade Block Production")
+
+	// Start op-nodes
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	// Verify Interop is not active at genesis yet
+	l2Head := actors.ChainA.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain A")
+	require.False(t, is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain A")
+
+	l2Head = actors.ChainB.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain B")
+	require.False(t, is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain B")
+
```

Comment:
nit, define `rollupConfigX` earlier and use here.

Reply:
Fixed in c6fd3b743583a4bdbc626c27c8e5b1a8af210d24.